VideoLocker/src/main/java/org/edx/mobile/model/api/CourseEntry.java

```diff
@@ -248,22 +252,79 @@ public boolean isGroupAvailable(SocialUtils.SocialType type) {
 
     }
 
-    public String getDescription(Context context){
+    public String getDescription(Context context, boolean withStartDate) {
         StringBuilder detailBuilder = new StringBuilder();
-        if ( getOrg() != null){
-            detailBuilder.append( getOrg());
+
+        if (!TextUtils.isEmpty(org)) {
+            detailBuilder.append(org);
         }
-        if ( getNumber() != null) {
-            if (detailBuilder.length() > 0){
+
+        if (!TextUtils.isEmpty(number)) {
+            if (detailBuilder.length() > 0) {
                 detailBuilder.append(" | ");
             }
-            detailBuilder.append( getNumber());
+            detailBuilder.append(number);
 
         }
-        //https://openedx.atlassian.net/browse/MA-858
-        //we remove the ending data for now
+
+        if (withStartDate) {
+            if (detailBuilder.length() > 0) {
+                detailBuilder.append(" | ");
+            }
+
+            detailBuilder.append(getFormattedStartDate(context));
+        }
+
+        if (detailBuilder.length() > 0) {
+            int secondlastCharIndex = detailBuilder.length() - 2;
+            if (detailBuilder.charAt(secondlastCharIndex) == '|') {
+                detailBuilder.delete(secondlastCharIndex, detailBuilder.length());
+            }
+        }
 
         return detailBuilder.toString();
     }
 
+    @SuppressLint("SimpleDateFormat")
+    public String getFormattedStartDate(Context context){
+        StringBuilder dateBuilder = new StringBuilder();
+
+        SimpleDateFormat dateFormat = new SimpleDateFormat("MMMM dd");
+        Date startDate = DateUtil.convertToDate(start);
+        Date endDate = DateUtil.convertToDate(end);
+        if (hasDefaultStartDate()) {
+            dateBuilder.append(context.getString(R.string.assessment_empty_coming_soon).toUpperCase());
+        } else if (isEnded()) {
+            if (endDate != null) {
+                dateBuilder.append(context.getString(R.string.label_ended));
+                dateBuilder.append(" - ");
+                dateBuilder.append(dateFormat.format(endDate));
+            }
+        } else if (isStarted()) {
+            if (endDate != null) {
+                dateBuilder.append(context.getString(R.string.label_ending_on));
+                dateBuilder.append(" - ");
+                dateBuilder.append(dateFormat.format(endDate));
+            }
+        } else {
+            if (startDate != null) {
+                dateBuilder.append(context.getString(R.string.label_starting_from));
+                dateBuilder.append(" - ");
+                dateBuilder.append(dateFormat.format(startDate));
```

Comment:
hyphen's don't necessarily translate to other languages. Also, this won't be reversed for RTL languages.

How about changing the strings to have a date placeholder instead of using an appender? E.g. `Ending - {date}`